Shared_AIHS2/Clothes/KoiClothesOverlayController.Hooks.cs

```diff
@@ -64,6 +67,70 @@ private static CmpClothes GetCustomClothesComponent(ChaControl chaControl, int k
             }
 
             #endregion
+
+
+            [HarmonyPostfix]
+            [HarmonyWrapSafe]
+            [HarmonyPatch(typeof(ChaControl), nameof(AIChara.ChaControl.InitBaseCustomTextureClothes))]
+            public static void InitBaseCustomTextureClothesPostHook(ChaControl __instance, int parts)
+            {
+                var updated = false;
+                var clothesId = GetClothesIdFromKind(true, parts);
+                clothesId = MakeColormaskId(clothesId);
+
+                var registration = CharacterApi.GetRegisteredBehaviour(typeof(KoiClothesOverlayController));
+                if (registration == null) throw new ArgumentNullException(nameof(registration));
+                foreach (var controller in registration.Instances.Cast<KoiClothesOverlayController>())
+                {
+                    for (int i = 0; i < 3; i++)
+                    {
+                        var tex = controller.GetOverlayTex(clothesId, false)?.Texture;
+                        if (tex != null)
+                        {
+                            if (parts < __instance.ctCreateClothes.GetLength(0) && i < __instance.ctCreateClothes.GetLength(1) && __instance.ctCreateClothes[parts, i] != null)
+                            {
+                                updated = true;
+                                __instance.ctCreateClothes[parts, i].SetTexture(ChaShader.ColorMask, tex);
+                            }
+                            if (parts < __instance.ctCreateClothesGloss.GetLength(0) && i < __instance.ctCreateClothesGloss.GetLength(1) && __instance.ctCreateClothesGloss[parts, i] != null)
+                            {
+                                updated = true;
+                                __instance.ctCreateClothesGloss[parts, i].SetTexture(ChaShader.ColorMask, tex);
+                            }
+                        }
+                    }
+                }
+                if (updated)
+                {
+                    // Since a custom color mask is now used, enable all color fields to actually make full use of it.
+                    var clothesComponent = __instance.GetCustomClothesComponent(parts);
+                    clothesComponent.useColorN01 = true;
+                    clothesComponent.useColorN02 = true;
+                    clothesComponent.useColorN03 = true;
+                    // Reflect changed UseColors 
+                    KoiClothesOverlayGui.RefreshMenuColors(parts);
+                }
+            }
+
+            public static Texture GetColormask(KoiClothesOverlayController controller, string clothesId)
+            {
+                if(GetKindIdsFromColormask(clothesId, out int? kind, out int? subKind))
+                {
+                    var listInfo = controller.ChaControl.infoClothes[(int)kind];
```

Comment:
This will crash when GetKindIdsFromColormask returns true but still has a null kind, see comment above.